proxy/envoy/config.go

```diff
@@ -410,7 +421,69 @@ func buildRoutes(svc *model.Service, rulesMap map[string][]*config.RouteRule,
 		routes = append(routes, Route{Prefix: "/", Cluster: OutboundClusterPrefix + svc.String()})
 	}
 
-	return routes
+	return routes, faultsByDestination
+}
+
+// buildFaultFilters builds a list of fault filters for the http route. If the route points to a single
+// cluster, an array of size 1 is returned. If the route points to a weighted cluster, an array of fault
+// filters (one per cluster entry in the weighted cluster) is returned.
+func buildFaultFilters(route *Route, faultRule *config.HttpFaultInjection) []Filter {
+	faults := make([]Filter, 0)
+	if route.WeightedClusters != nil {
+		for _, cluster := range route.WeightedClusters.Clusters {
+			faults = append(faults, buildFaultFilter(cluster.Name, faultRule))
+		}
+	} else {
+		faults = append(faults, buildFaultFilter(route.Cluster, faultRule))
+	}
+	return faults
+}
+
+// buildFaultFilter builds a single fault filter
+func buildFaultFilter(internalUpstreamName string, faultRule *config.HttpFaultInjection) Filter {
+	return Filter{
+		Type: "decoder",
+		Name: "fault",
+		Config: FilterFaultConfig{
+			UpstreamCluster: internalUpstreamName,
+			Headers:         buildHeaders(faultRule.Headers),
+			Abort:           buildAbortConfig(faultRule.Abort),
+			Delay:           buildDelayConfig(faultRule.Delay),
+		},
+	}
+}
+
+// buildAbortConfig builds the envoy config related to abort spec in a fault filter
+func buildAbortConfig(abortRule *config.HttpFaultInjection_Abort) *AbortFilter {
+	if abortRule == nil {
+		return nil
+	}
+	if abortRule.GetHttpStatus() == 0 {
+		// We do not support gRPC and http2 errors now.
+		return nil
+	}
+
+	return &AbortFilter{
+		Percent:    int(abortRule.Percent),
```

Comment:
please test it with a float and an int value. we do several translations and JS is funky about numbers.